kcidb/misc.py

```diff
@@ -165,6 +165,27 @@ def non_negative_int(string):
     return int(string)
 
 
+def schema_versions(value):
+    """
+    Matches the argparse type function interface.
+
+    Args:
+        value: The value to validate.
+
+    Returns:
+        The schema version.
+
+    Raises:
+        argparse.ArgumentTypeError: the value is not a schema version.
+    """
+    if value == 'LATEST':
+        return value
+    if not re.match("^[1-4]+$", value):
+        raise argparse.ArgumentTypeError(
+            f'{repr(value)} is not a previous schema version')
+    return "V"+value
```

Comment:
Nice job figuring out how to do this, but let's do it the simple way first.
Just add the argument definition inside the `schema_main()`.

You can build a list of available schema versions using a loop like this:
```python
schema_major_versions = []
version = kcidb_io.schema.LATEST
while version:
    schema_major_versions.append(version.major)
    version = version.previous
```
Then you can use it to define available choices for the argument values. We can start with the absent argument meaning the latest version, without an explicit "LATEST" value. Also, no need to require the "V" to be present there, just the number is enough and creates less hassle.

Once you get your `major` version value from the argument, you can lookup the schema like this:
```python
version = kcidb_io.schema.LATEST
while version and version.major != major:
    version = version.previous
if not version:
    raise Exception("No version found for major number {major}")
```